index.test.js

```diff
@@ -0,0 +1,67 @@
+document.body.innerHTML = `<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>To Do List In JavaScript</title>
+    <link
+      rel="stylesheet"
+      href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.8.3/font/bootstrap-icons.css"
+    />
+    <link rel="stylesheet" href="../src/style.css" />
+  </head>
+  <body>
+    <div class="container">
+      <header>
+        <div class="new-todo">
+          <form id="todoform">
+            <input
+              type="text"
+              name="newtodo"
+              id="newtodo"
+              placeholder="Type here and press Enter ..."
+            />
+          </form>
+          <button type="button" class="deletebutton" id="removeitem">
+            Clear all completed
+          </button>
+        </div>
+      </header>
+      <div id="todos-list"></div>
+      <div class="notification"></div>
+    </div>
+    <script src="/dist/awesome.js"></script>
+    <script src="/src/index.js" type="module" defer></script>
+  </body>
+</html>
+`;
+const { todos, saveTodo, deleteTodo } = require('./utils.js');
+
+const todoInput = document.getElementById('newtodo');
+
+describe('Add Task Test ', () => {
+  test('Add task', () => {
+    todoInput.value = 'Task 1';
+    saveTodo();
+    expect(todos).toHaveLength(1);
+  });
+  test('Add task 2', () => {
+    todoInput.value = 'Task 2';
+    saveTodo();
+    expect(todos).toHaveLength(2);
+  });
+  test('Add task 3', () => {
+    expect(todos.length).toEqual(2);
+  });
+});
+
+describe('remove Task Test ', () => {
+  test('remove task', () => {
+    const newTodos = deleteTodo(0);
+    expect(newTodos).toHaveLength(1);
+  });
+  test('remove task', () => {
+    const newTodos = deleteTodo(0);
+    expect(newTodos).toHaveLength(0);
+  });
+});
```

Comment:
- [ ] **[OPTIONAL]:** You did a great job writing your tests! 💯 However, I suggest you not only test against your `"todos"` variable, which interacts with your localStorage, but test the DOM interaction too. For example, you could select your `todos-list` element and assert the length of its children after adding and removing a to-do. When you add one, you expect this list to have one more child, and when you remove one, you expect it to have one less. Adding this basic test gives you the confidence to say that your code is interacting well with the LocalStorage API, and with the DOM too. 🤓 🛠️